jina/executors/indexers/keyvalue.py

```diff
@@ -69,12 +70,58 @@ def add(self, keys: Iterator[int], values: Iterator[bytes], *args, **kwargs):
             # print(f'l: {l} p: {p} r: {r} r+l: {r + l} size: {self._size}')
 
     def query(self, key: int) -> Optional[bytes]:
+        # FIXME this shouldn't be required
+        if self.query_handler is None:
+            self.query_handler = self.get_query_handler()
```

Comment:
```suggestion

```

Reply:
Yes, but it fails sometimes. Need to debug it more and figure out why